base/libgit2/libgit2.jl

```diff
@@ -424,9 +424,11 @@ function merge!(repo::GitRepo;
                 remotename = with(GitConfig, repo) do cfg
                     LibGit2.get(String, cfg, "branch.$branchname.remote")
                 end
-                with(GitReference(repo, "refs/remotes/$remotename/$branchname")) do ref
-                    obj = LibGit2.Oid(ref)
-                    LibGit2.create_branch(repo, branchname, get(GitCommit, repo, obj))
+                obj = with(GitReference(repo, "refs/remotes/$remotename/$branchname")) do ref
+                    LibGit2.Oid(ref)
+                end
+                with(get(GitCommit, repo, obj)) do cmt
```

Comment:
have you been able to come up with any scheme or logic for when and what needs management like this?